code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -47,6 +47,12 @@
 		if(C.can_inject(null, 0, inject_target, 0))
 			C.reagents.add_reagent("spidertoxin", venom_per_bite)
 
+/mob/living/simple_animal/hostile/poison/giant_spider/New()
+	..()
+	spawn(1)
+	if(!faction.len)

```

Comment:
This happens any time spiderlings are spawned manually without eggs being laid. I have tried setting the faction in other places in the spider life cycle but nothing I did changed the faction of the resulting spider, for some reason. It's much easier to do it this way, by checking the actual spider once it has spawned.

Reply:
It's because the only place where a spider's faction is defined, is on a giant spider mob. An eggcluster then inherits the faction of the egg-layer, then it gives it out to hatched spiderling...

This problem also appears for the spiderman gland.

Have you tried setting the spiderling's faction when the spiderling is spawned on the changeling or spiderman gland?

Reply:
Yes. I've tried setting the faction directly in the changeling's ability for example, didn't change a thing.

Reply:
Remove this band-aid until you could fix the root cause of the problem